qc_opendrive/main.py

```diff
@@ -71,21 +59,87 @@ def execute_checker(
             "Preconditions are not satisfied. Skip the check.",
         )
 
-        return
+        return False
 
-    # Checker definition setting. If not satisfied then set status as SKIPPED and return
-    if required_definition_setting:
-        schema_version = checker_data.schema_version
 
-        splitted_rule_uid = checker.RULE_UID.split(":")
-        if len(splitted_rule_uid) != 4:
-            raise RuntimeError(f"Invalid rule uid: {checker.RULE_UID}")
+def check_version(checker: types.ModuleType, checker_data: models.CheckerData) -> bool:
+    """
+    Check definition setting and applicable version.
+    If not satisfied then set status as SKIPPED or ERROR and return False
+    """
+    schema_version = checker_data.schema_version
 
-        definition_setting = splitted_rule_uid[2]
-        if (
-            schema_version is None
-            or utils.compare_versions(schema_version, definition_setting) < 0
-        ):
+    splitted_rule_uid = checker.RULE_UID.split(":")
+    if len(splitted_rule_uid) != 4:
+        raise RuntimeError(f"Invalid rule uid: {checker.RULE_UID}")
+
+    definition_setting = splitted_rule_uid[2]
+    definition_setting_expr = f">={definition_setting}"
+    match_definition_setting = version.match(schema_version, definition_setting_expr)
+
+    applicable_version = (
+        checker.APPLICABLE_VERSION if hasattr(checker, "APPLICABLE_VERSION") else None
+    )
+
+    # Check whether applicable version specification is valid
+    if applicable_version is not None and not version.is_valid_version_expression(
+        applicable_version
+    ):
```

Comment:
> with "", it's an invalid version but valid definition for users

Ouch you are right! `"".split(",") == [""] != []`. My fault, and **good catch!**.

To have both `None` and `""` idempotent we need to parse `""` as a clauses list `[]`. This can be done when the split is applied with a `- [""]` on the result of the split (or `.remove("")`).

This will make `""` a valid version (idempotent to undefined) and a valid definition for a developer (which can define it even when not needed). This will also lower cyclomatic complexity of the current file.

I would also add a test on `None` and `""` in unit-test.